airbyte-cdk/bulk/toolkits/extract-cdc/src/main/kotlin/io/airbyte/cdk/read/cdc/DebeziumStateFilesAccessor.kt

```diff
@@ -90,21 +76,20 @@ class DebeziumStateFilesAccessor : AutoCloseable {
     }
 
     fun readSchema(): DebeziumSchemaHistory {
-        fileSchemaHistory.start()
-        val schema: List<HistoryRecord> = buildList {
-            recoverRecords(fileSchemaHistory, Consumer(this::add))
+        var schema: List<HistoryRecord> = listOf()
+        doWithFileSchemaHistory(schemaFilePath) { fileSchemaHistory ->
+            schema = buildList { recoverRecords(fileSchemaHistory, Consumer(this::add)) }
         }
-        fileSchemaHistory.stop()
+
         return DebeziumSchemaHistory(schema)
     }
 
     fun writeSchema(schema: DebeziumSchemaHistory) {
-        fileSchemaHistory.initializeStorage()
-        fileSchemaHistory.start()
-        for (r in schema.wrapped) {
-            storeRecord(fileSchemaHistory, r)
```

Comment:
This is where the problem was: In addition to disk, each record is also stored in `fileSchemaHistory.records`